### ISSUE TITLE
Huggingface hosted inference api: rogue character in mask, also perhaps wrong task?

### ISSUE THREAD

**Issue by oconnoat** [JohnGiorgi/DeCLUTR#147]
The Declutr work is very exciting, thank you for sharing.

Issue part one: The huggingface hosting site for the model has the default fill-text task - perhaps a note that this is not a relevant task in the model card?

Additionally, the code seems to have an error, where the results include a rogue character at the start of the decoded mask token. 
<img width="823" alt="Screen Shot 2020-08-25 at 09 42 54" src="https://user-images.githubusercontent.com/3483698/91207357-7ab34d80-e6bd-11ea-872a-b5d2cb6f7b13.png">

Running the raw pipeline code does not seem to reproduce this error:
```
from transformers import AutoModelWithLMHead, AutoTokenizer
import torch
import unicodedata

lmtokenizer = AutoTokenizer.from_pretrained("johngiorgi/declutr-small")
lmmodel = AutoModelWithLMHead.from_pretrained("johngiorgi/declutr-small")

sequence = f"The capital of France is {tokenizer.mask_token}."

input = lmtokenizer.encode(sequence, return_tensors="pt")
mask_token_index = torch.where(input == lmtokenizer.mask_token_id)[1]

token_logits = lmmodel(input)[0]
mask_token_logits = token_logits[0, mask_token_index, :]

top_5_tokens = torch.topk(mask_token_logits, 5, dim=1).indices[0].tolist()

for token in top_5_tokens:
    print(lmtokenizer.decode([token]).encode('utf-8'))
    print([unicodedata.name(c) for c in lmtokenizer.decode([token])])
    print(sequence.replace(lmtokenizer.mask_token, lmtokenizer.decode([token])))
```

and a similar error seems to propagate here https://huggingface.co/xlm-roberta-large?text=The+goal+of+life+is+%3Cmask%3E. perhaps this is an issue for Hugging face?

**Comment by JohnGiorgi** [comment]
Thanks for the kind words.

I am not so sure this is an error. "Ġ" is not a "rouge" character, but an encoded space character of the BPE tokenization that these models use (see [here](https://github.com/huggingface/transformers/issues/3867)). You will notice that the `fill-mask` task of `roberta-base` [model card also generates them](https://huggingface.co/roberta-base?text=Paris+is+the+%3Cmask%3E+of+France.). Furthermore, if you use the `roberta-base` tokenizer from HuggingFace, there are included in the output:

```python

tokenizer = AutoTokenizer.from_pretrained("roberta-base")                                                                                                                                                                                 

tokenizer.tokenize("Paris is the capital of France")                                                                                                                                                                                      
['Paris', 'Ġis', 'Ġthe', 'Ġcapital', 'Ġof', 'ĠFrance']
```

Please let me know if I have misunderstood anything.

**Comment by oconnoat** [comment]
Looks like you're right! It is a hf issue.
Thanks for looking into it.
